cli/integration/tests/waiter/test_cli.py

```diff
@@ -2352,3 +2352,97 @@ def test_stop_ping_service_force_kill(self):
 
     def test_stop_ping_service_no_kill(self):
         self.run_maintenance_start_test(cli.stop, start_args='--no-kill', ping_token=True)
+
+    def __test_signal(self, get_possible_instances_fn, signal_type=None, signal_flags=None, stdin=None,
+                      min_instances=1, multiple_services=False, is_failed_instance=False,
+                      test_service=False, test_instance=False):
+        token_name = self.token_name()
+        token_fields = util.minimal_service_description()
+        token_fields['min-instances'] = min_instances
+        if is_failed_instance:
+            token_fields['cmd'] = 'this_is_an_invalid_command'
+        try:
+            if multiple_services:
+                token_new_fields = util.minimal_service_description()
+                util.post_token(self.waiter_url, token_name, token_new_fields)
+                util.ping_token(self.waiter_url, token_name)
+            util.post_token(self.waiter_url, token_name, token_fields)
+            service_id = util.ping_token(self.waiter_url, token_name,
+                                         expected_status_code=503 if is_failed_instance else 200)
+            if is_failed_instance:
+                goal_fn = lambda insts: 0 < len(insts['failed-instances']) and \
+                                        0 == len(insts['killed-instances'])
+            else:
+                goal_fn = lambda insts: min_instances == len(insts['active-instances']) and \
+                                        0 == len(insts['failed-instances']) and \
+                                        0 == len(insts['killed-instances'])
+            util.wait_until_routers_service(self.waiter_url, service_id, lambda service: goal_fn(service['instances']))
+            instances = util.instances_for_service(self.waiter_url, service_id)
+            possible_instances = get_possible_instances_fn(service_id, instances)
+            signal_flags = [signal_flags] if signal_flags else []
+            if test_instance:
+                possible_instances = possible_instances[0:1]
+                signal_dest = possible_instances[0]['id']
+                signal_flags.append('-i')
+            elif test_service:
+                signal_dest = service_id
+                signal_flags.append('-s')
+            else:
+                signal_dest = token_name
+            logging.info(f'{signal_dest} is id used!')
+            cp = cli.signal(self.waiter_url, signal_type, signal_dest, signal_flags=' '.join(signal_flags), stdin=stdin)
+            stdout = cli.stdout(cp)
+            self.assertEqual(0, cp.returncode, cp.stderr)
+            self.assertIn(f'Successfully sent {signal_type} to', stdout)
```

Comment:
Please add an assertion to find killed instances (in wait until mechanism as there may be a delay in killed instances being registered) and confirm it matches the instance ID.
Separately, you can also query the `/apps` endpoint and confirm that active instances does not contain the killed instance.